upcloud/managed_database.go

```diff
@@ -0,0 +1,693 @@
+	// IPv4 hosts.
+	ManagedDatabaseAllIPv4 = "0.0.0.0/0"
+	// ManagedDatabasePropertyPublicAccess enables public access via internet to the service. A separate public
+	// endpoint DNS name will be available under Components after enabling.
+	ManagedDatabasePropertyPublicAccess = "public_access"
+)
+
+// User types
+const (
+	// ManagedDatabaseUserTypePrimary is a type of the primary user of a managed database service. There can be only
+	// one primary user per service. The primary user has administrative privileges to manage logical databases and
+	// users thru the database's native API.
+	ManagedDatabaseUserTypePrimary = "primary"
+	// ManagedDatabaseUserTypeNormal is a type of a normal database user of a managed database service. There can
+	// be multiple normal users and the primary user can manage the privileges of these users thru the database's
+	// native API.
+	ManagedDatabaseUserTypeNormal = "normal"
+)
+
+// Authentication types
+const (
+	// ManagedDatabaseUserAuthenticationCachingSHA2Password selects "caching_sha2_password" type of authentication type.
+	// This type is only supported with MySQL services.
+	//nolint:gosec
+	ManagedDatabaseUserAuthenticationCachingSHA2Password = "caching_sha2_password"
+	// ManagedDatabaseUserAuthenticationMySQLNativePassword selects "mysql_native_password" type of authentication type.
+	// This type is only supported with MySQL services.
+	//nolint:gosec
+	ManagedDatabaseUserAuthenticationMySQLNativePassword = "mysql_native_password"
+)
+
+// ManagedDatabase represents an existing managed database instance
+type ManagedDatabase struct {
+	Backups          []ManagedDatabaseBackup         `json:"backups"`
+	Components       []ManagedDatabaseComponent      `json:"components"`
+	CreateTime       time.Time                       `json:"create_time"`
+	Maintenance      ManagedDatabaseMaintenance      `json:"maintenance"`
+	Name             string                          `json:"name"`
+	NodeCount        int                             `json:"node_count"`
+	NodeStates       []ManagedDatabaseNodeState      `json:"node_states"`
+	Plan             string                          `json:"plan"`
+	Powered          bool                            `json:"powered"`
+	Properties       ManagedDatabaseProperties       `json:"properties"`
+	State            string                          `json:"state"`
+	Title            string                          `json:"title"`
+	Type             string                          `json:"type"`
+	UpdateTime       time.Time                       `json:"update_time"`
+	ServiceURI       string                          `json:"service_uri"`
+	ServiceURIParams ManagedDatabaseServiceURIParams `json:"service_uri_params"`
+	Users            []ManagedDatabaseUser           `json:"users"`
+	UUID             string                          `json:"uuid"`
+	Zone             string                          `json:"zone"`
+}
+
+// ManagedDatabaseBackup represents a taken full backup's point in time
```

Comment:
This comment could use some editing; `.. represents a full backup taken at a point in time` or something?

Reply:
whoops :)

Reply:
implemented